### DISCUSSION TITLE
I'm having trouble installing (I'm a n00b)

### DISCUSSION THREAD

**Discussion by ShiftlessDrifter** [younesaassila/ttv-lol-pro#192] [Q&A] [answered]
I'm using the latest version of Firefox on Windows 10. I'm new to github and find this site very confusing. I had a previous version of TTV LOL back when it was still available from Mozilla. Anyway, onto the problem:

I first uninstalled the extension in case there were problems with the newer one. After coming to this site I went to the Code dropdown menu and downloaded the zip file. Then after reading the instructions I saw there is a releases page for a specific version for Firefox and it's a different zip file. (So now I have two zip files but I'm going to ignore the first one). 

I continued following the instructions for installing it and I think it was installed correctly; but how can I check to see if I have the latest version installed? I am still getting ads on Twitch. I know this extension only blocks some ads, but I'm still not sure if I installed it correctly.

Second, after closing my browser and opening it up again, the extension is gone. (I guess this is what "Temporary" means?) I was able to "Undo" the removal of the original extension but I'm just confused as to what to do, and how to keep this installed permanently. Or do I have to install it every time I load up Firefox?

I wish there was a guide for complete morons like me because I don't understand this website at all, or the installation process for this extension.

**Comment by thinkpad4** [comment]
@ShiftlessDrifter You need the Developer version of Firefox to have it stay permanently `https://www.mozilla.org/firefox/developer/`

**Reply by ShiftlessDrifter** [reply]
This worked for me, thank you.

**Comment by sn-o-w** [comment]
@ShiftlessDrifter @thinkpad4 You can patch Firefox with this: https://github.com/xiaoxiaoflood/firefox-scripts#readme

There's even a YouTube video that shows the necessary steps if it's too complicated.

After you installed userChromeJS, you need to go to `about:addons` and just manually install the Firefox zip file you download from [here](https://github.com/younesaassila/ttv-lol-pro/releases/latest).

I tested it a few days ago, installing TTV LOL PRO v2 on Firefox 114. It works like a charm.

**Comment by sn-o-w** [comment, marked as answer]
**LATER EDIT**: I've simplified that thing to only allow installing unsigned add-ons, nothing more than that. 😊

So you can ignore my previous comment and proceed to this:

[install-unsigned-addons.zip](https://github.com/younesaassila/ttv-lol-pro/files/11780247/install-unsigned-addons.zip)

- Close Firefox. 
- Download the above zip archive, extract its content to `C:\Program Files\Mozilla Firefox`
- Open Firefox, go to `about:addons#extensions` and manually install any unsigned addon you want.

**Reply by thinkpad4** [reply]
@sn-o-w That worked PERFECT!  Thank you very much, and it's easy to remove when/if Mozilla ever approves ttvlolpro for firefox

**Reply by sn-o-w** [reply]
@thinkpad4 You're welcome. This thing should work with newer Firefox releases. I'm curious if files survive past the new Firefox updates. I guess we'll see when Firefox 115 releases. 😂

**Reply by sn-o-w** [reply]
**UPDATE**: This should work on Firefox v115+.

[install-unsigned-addons-v2.zip](https://github.com/younesaassila/ttv-lol-pro/files/12440878/install-unsigned-addons-v2.zip)

Same installation steps.

**Comment by jumphil** [comment]
Hopefully we can get an xpi file or a listing on the Firefox Addons site. I'm still using the 1.9.3 xpi for now which works on the non-developer version, permanently installed.

**Reply by sn-o-w** [reply]
@jumphil Check out the file I posted. You can patch Firefox and you can install v2 of the addon. 😎

**Comment by younesaassila** [comment]
Mozilla has finally approved the extension! https://addons.mozilla.org/addon/ttv-lol-pro/
